pkg/system/resources.go

```diff
@@ -1,76 +1,73 @@
-	buildkitdServiceCPURequest    = *resource.NewMilliQuantity(200, resource.DecimalSI) // BUILDKITD_SERVICE_CPU_REQUEST
-)
+		req, limit, both := strings.Cut(rs, ":")
+		if !both {
+			_, err := resource.ParseQuantity(rs)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
 
-func RegistryResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_REQUEST", registryMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("REGISTRY_CPU_REQUEST", registryCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("REGISTRY_MEMORY_LIMIT", registryMemoryLimit),
-		},
+		_, err := resource.ParseQuantity(req)
+		if err != nil {
+			return err
+		}
+		_, err = resource.ParseQuantity(limit)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
 }
 
-func BuildkitdResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_MEMORY_REQUEST", buildkitdMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("BUILDKITD_CPU_REQUEST", buildkitdCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_MEMORY_LIMIT", buildkitdMemoryLimit),
-		},
-	}
-}
+func ResourceRequirementsFor(memory, cpu string) corev1.ResourceRequirements {
+	requirements := corev1.ResourceRequirements{Requests: corev1.ResourceList{}, Limits: corev1.ResourceList{}}
 
-func BuildkitdServiceResources() corev1.ResourceRequirements {
-	return corev1.ResourceRequirements{
-		Requests: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_SERVICE_MEMORY_REQUEST", buildkitdServiceMemoryRequest),
-			corev1.ResourceCPU:    envOrDefault("BUILDKITD_SERVICE_CPU_REQUEST", buildkitdServiceCPURequest),
-		},
-		Limits: corev1.ResourceList{
-			corev1.ResourceMemory: envOrDefault("BUILDKITD_SERVICE_MEMORY_LIMIT", buildkitdServiceMemoryLimit),
-		},
+	var resources = map[corev1.ResourceName]string{
+		corev1.ResourceMemory: memory,
+		corev1.ResourceCPU:    cpu,
 	}
-}
 
-func envOrDefault(env string, def resource.Quantity) resource.Quantity {
-	if env = os.Getenv(env); env == "" {
-		return def
-	}
+	for resourceType, resourceString := range resources {
+		reqString, limitString, both := strings.Cut(resourceString, ":")
+
+		if !both {
+			parsedReq, err := resource.ParseQuantity(resourceString)
+			if err == nil {
+				requirements.Requests[resourceType] = parsedReq
+			}
+			continue
+		}
 
-	quantity, err := resource.ParseQuantity(env)
-	if err == nil {
-		return quantity
+		parsedReq, err := resource.ParseQuantity(reqString)
+		if err == nil {
+			requirements.Requests[resourceType] = parsedReq
+		}
+
+		parsedLimit, err := resource.ParseQuantity(limitString)
+		if err == nil {
+			requirements.Limits[resourceType] = parsedLimit
+		}
```

Comment:
Just want to make sure that we are intentionally ignoring all errors here.

Reply:
It's a good thing to call out. My thought process was that when this is called its likely in areas that cannot/should not error. For example, this area of code.

https://github.com/tylerslaton/runtime/blob/ed7fae64670fc4907a7fd006fabe3c96955cb704/pkg/imagesystem/buildertemplate.go#L58-L59

This makes it so that we don't `panic` in these situations but I would like your opinion here. The main scenario this presents as an annoyance is if the user manually edits or creates an `apiv1.Config` and provides an invalid value. In this situation they would receive an `0` amount.

Reply:
Hmm...I think it is okay to leave it as is :+1: